File: backend/src/main/java/com/apocalypse/thefall/repository/item/ItemInstanceRepository.java
package com.apocalypse.thefall.repository.item;

import com.apocalypse.thefall.entity.instance.ItemInstance;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface ItemInstanceRepository extends JpaRepository<ItemInstance, Long> {
}

File: backend/src/main/java/com/apocalypse/thefall/repository/iteminstance/ItemInstanceRepository.kt
package com.apocalypse.thefall.repository.iteminstance

import com.apocalypse.thefall.entity.instance.ItemInstance
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Repository

@Repository
interface ItemInstanceRepository : JpaRepository<ItemInstance, Long> {
    @Query(
        """
        select distinct ii
        from ItemInstance ii
        left join fetch ii.item it
        where ii.inventory.id = :inventoryId
    """
    )
    fun loadForInventoryWithItem(inventoryId: Long): List<ItemInstance>
}